crypto/rand/rand_unix.c

```diff
@@ -348,50 +381,34 @@ static uint64_t get_time64(void)
         read_wall_time(&t, TIMEBASE_SZ);
         return TWO32TO64(t.tb_high, t.tb_low);
     }
-# elif defined(OSSL_POSIX_TIMER_OKAY)
+# else
+
+#  if defined(OSSL_POSIX_TIMER_OKAY)
     {
         struct timespec ts;
-        clockid_t cid;
 
-#  ifdef CLOCK_BOOTTIME
-        cid = CLOCK_BOOTTIME;
-#  elif defined(_POSIX_MONOTONIC_CLOCK)
-        cid = CLOCK_MONOTONIC;
-#  else
-        cid = CLOCK_REALTIME;
-#  endif
+#   ifdef CLOCK_BOOTTIME
+#    define CLOCK_TYPE CLOCK_BOOTTIME
+#   elif defined(_POSIX_MONOTONIC_CLOCK)
+#    define CLOCK_TYPE CLOCK_MONOTONIC
+#   else
+#    define CLOCK_TYPE CLOCK_REALTIME
+#   endif
 
-        if (clock_gettime(cid, &ts) == 0)
+        if (clock_gettime(CLOCK_TYPE, &ts) == 0)
```

Comment:
BTW: The manual page says that the call only fails if the pointer is out of range or the clock type is not supported. So this call should never fail. Is it really necessary to check the return value?